feat(chat): add settings to alter the chat view's base font

antis81 opened the pull request:
Closes #672
- [x] Introduce `Chat/chatMessageFont` settings value
  -  Defaults to GUI style font according to previous behavior.
- [x] Add Settings UI
- [x] Cleanup (chat) settings UI
- [x] Set the font censequently as the chat's base font.
  - Author, message and timestamp columns are based on a single font now. I can separate those to set a font on each column, but actually I think it is totally fine as is (See screenshots below).
# Settings UI:

![qtox-chat-base-font-settings-ui](https://cloud.githubusercontent.com/assets/440517/16413604/2f91b3c4-3d33-11e6-8e57-6901425e31c7.png)

**Note:** @zetok I moved both text format rows to the top of the group box and shorten the markdown label to something like "Markdown format" (the gap makes the dialog look like a mixed fruit salad :smile:)
# Settings UI after cleanup:

![qtox-chat-settings-ui](https://cloud.githubusercontent.com/assets/440517/16422139/95186b6a-3d57-11e6-902b-552da5bf280d.png)
# Example 1 (from WIP interims-state) for chat view with "Comic Sans MS" font (size: 18px):

![qtox-chat-base-font](https://cloud.githubusercontent.com/assets/440517/16413496/947b29ce-3d32-11e6-8409-e2fb91100f81.png)
# Example 2 (after) -> consistent appearance with same base font!

![qtox-chat-base-font-complete](https://cloud.githubusercontent.com/assets/440517/16419190/1ffd63da-3d4d-11e6-9f36-6559db7c9078.png)

antis81 committed:
feat(chat): add settings to alter the chat view's base font

Defaults to GUI style font according to previous behavior.

antis81 committed:
feat(chat): add the ui settings to alter font and size for chat messages

sudden6 commented on src/widget/form/settings/generalform.cpp:
```diff
@@ -109,6 +110,8 @@ GeneralForm::GeneralForm(SettingsWidget *myParent) :
     bodyUI = new Ui::GeneralSettings;
     bodyUI->setupUi(this);
 
+    Settings& s = Settings::getInstance();
```
would be neat to make the choice between `s` or `Settings::getInstance()` consistent throughout the method.

antis81 replied:
Done.

zetok commented on src/chatlog/content/text.cpp:
```diff
@@ -247,9 +250,12 @@ void Text::regenerate()
+            doc->setDefaultStyleSheet(defStyleSheet);
             doc->setHtml(text);
-        else
+        } else {
```
style should be 

``` c++
}
else
{
```

antis81 committed:
feat(chat): consistent use of the chat font settings

Note: Currently a CSS stylesheet has to be set, resembling the font settings for the HTML document, that is used to render chat messages.

antis81 committed:
refactor(chatform): use reference var when initializing from settings

antis81 commented:
@zetok It's ready!
